src/openforms/submissions/tasks/__init__.py

```diff
@@ -22,128 +21,59 @@
-    so the :func:`on_completion` handler must kick this off AND have its own task ID
-    that finishes, which is checked in the submission status endpoint.
-    """
-    send_confirmation_email_task = maybe_send_confirmation_email.si(submission_id)
-    send_confirmation_email_task.delay()
+    # Send emails
+    schedule_emails_task = schedule_emails.si(submission_id)
 
+    # Hash attributes
     hash_identifying_attributes_task = maybe_hash_identifying_attributes.si(
         submission_id
     )
-    hash_identifying_attributes_task.delay()
-
-
-def on_completion_retry(submission_id: int) -> chain:
-    """
-    Celery chain of tasks to execute on a submission completion processing retry.
-
-    This differs from :func:`on_completion` in that it has a different "starting point"
-    and invokes some extra tasks or skips other tasks. It focuses on tasks that
-    typically fail downstream in external systems outside of our own control,
-    such as registration backends, appointment booking systems.
 
-    .. note::
-
-        The retry workflow may only need to execute a part of the entire flow, but
-        we still consider this an atomic unit/entrypoint to manage the dependencies
-        properly. It's important that the individual celery tasks making up the
-        workflow are idempotent and exit successfully when nothing needs to be done!
+    actions_chain = chain(
+        register_appointment_task,
+        pre_registration_task,
+        generate_report_task,
+        register_submission_task,
+        payment_status_update_task,
+        schedule_emails_task,
+        hash_identifying_attributes_task,
+    )
 
-    TODO: the results should be forgotten as part of the retry flow to not flood the
-    result backend!
+    async_result: AsyncResult = actions_chain.delay()
 
-    TODO: see if we can find a way to suppress exceptions from being sent to error
-    monitoring _if and only if_ they're part of this particular workflow.
-    """
-    pre_register_submission_task = pre_registration.si(submission_id)
-    register_submission_task = register_submission.si(submission_id).set(
-        ignore_result=True
-    )
-    update_payments_task = update_submission_payment_status.si(submission_id).set(
-        ignore_result=True
-    )
-    finalize_completion_retry_task = finalize_completion_retry.si(submission_id).set(
-        ignore_result=True
-    )
+    # obtain all the task IDs so we can check the state later
+    task_ids = async_result.as_list()
 
-    retry_chain = chain(
-        pre_register_submission_task,
-        register_submission_task,
-        update_payments_task,
-        finalize_completion_retry_task,
+    # NOTE - this is "risky" since we're running outside of the transaction (this code
+    # should run in transaction.on_commit)!
+    PostCompletionTasksIDs.objects.create(
+        tasks_ids=task_ids, submission=Submission.objects.get(id=submission_id)
```

Comment:
We can't do the same as we were doing before:
```python
Submission.objects.filter(id=submission_id).update(on_completion_task_ids=task_ids)
```
Since multiple events can trigger this chain of tasks. This would lead to race conditions where the tasks IDs are overwritten

Reply:
Was this the reason `on_completion_task_ids` was moved to a separate model?

Reply:
Yep